modules/openapi-generator/src/main/resources/python/python-experimental/model_utils.mustache

```diff
@@ -65,31 +65,53 @@ class OpenApiModel(object):
         # pick a new schema/class to instantiate because a discriminator
         # propertyName value was passed in
 
+        # Build a list containing all oneOf and anyOf descendants.
+        oneof_anyof_classes = cls._composed_schemas.get('oneOf', ()) +
+                                cls._composed_schemas.get('anyOf', ())
+        if oneof_anyof_classes and none_type in oneof_anyof_classes and args[0] is None:
+            # The input data is the 'null' value AND one of the oneOf/anyOf children
+            # is the 'null' type (which is introduced in OAS schema >= 3.1).
+            return None
+
         visited_composed_classes = kwargs.get('_visited_composed_classes', ())
         if (
             cls.discriminator is None or
             cls in visited_composed_classes
         ):
-            # we don't have a discriminator
-            # or we have already visited this class before and are sure that we
-            # want to instantiate it this time
+            # There is no discriminator.
+            # Or we have already visited this class before and are sure that we
+            # want to instantiate it this time. For example, this happens when
+            # we deserialize Dog from Animal. Dog contains an Animal instance. We go from Animal to Dog back to Animal
```

Comment:
For example if Animal has a discriminator
petType and we pass in "Dog", and the class Dog
allOf includes Animal, we move through Animal
once using the discriminator, and pick Dog.
Then in Dog, we will make an instance of the
Animal class but this time we won't travel
through its discriminator because we passed in
_visited_composed_classes = (Animal,)